### ISSUE TITLE
404 when using Jam for first time on Umbrel - "Unhandled Thrown Error!"

### ISSUE THREAD

**Issue by librethanks** [joinmarket-webui/jam#546]
**Expected behavior**

Launch the app, enter authentication, either skip intro or click through, create new wallet.

**Actual behavior**

Launch the app, enter authentication, either skip intro or click through, then:

"Unhandled Thrown Error!
404 Not Found

💿 Hey developer 👋

You can provide a way better UX than this when your app throws errors by providing your own errorElement props on < Route >"

**Steps to reproduce the problem**
  1. Open Jam app on Umbrel on either Tor or local network "umbrel.local"
  2. Skip intro or click through intro
  3. 404 error shows up

**Specifications**
  - Version: 0.1.1
  - Platform: Umbrel 0.5.1
  - Browser: Tor Browser 11.5.4 (based on Mozilla Firefox 91.13.0esr) (64-bit) and Firefox 105.0.3 (64-bit)

**Additional context**

This issue occurs on local network and over Tor browser. This issue continues after reinstalling the app.

![image](https://user-images.githubusercontent.com/97930095/196093253-b2870ecc-1282-4eb6-8cf0-fd7a4523b080.png)

**Comment by theborakompanioni** [comment]
Hey @librethanks, thank you very much for the detailed report!

This is unfortunately the current error message when the backend is unreachable in `v0.1.1`. It is fixed with https://github.com/joinmarket-webui/jam/pull/540 and will be included in `v0.1.2` (not released yet). However, this just means that the error message will change from the generic message you see now to a more descriptive message like `No connection to backend: $reason`.

Does the error persist? e.g. Do you still see the error after a system reboot?
If the errors persists, would you be able to share the latest lines of your log file?
(Please make sure it does not contain more information that necessary to identify the problem)
See here: https://jamdocs.org/FAQ/#how-do-i-view-the-log-file

**Comment by librethanks** [comment]
I've looked at the jamdocs page and it doesn't seem to point me in the right direction. I am able to ssh into my Umbrel node and grab the logs from the different docker containers, but there is no /var/log/jam/ directory or /home/<user>/.joinmarket/logs directory.

Here are the logs when I run `docker <container id for jam_jam_1>  logs` :

`[WARN  tini (8)] Tini is not running as PID 1 and isn't registered as a child subreaper.
Zombie processes will not be re-parented to Tini, so zombie reaping won't work.
To fix the problem, use the -s option or set the environment variable TINI_SUBREAPER to register Tini as a child subreaper, or run Tini as PID 1.
Remove leftover wallet lockfiles before startup...
Creating wallet jam_default if missing...
Loading wallet jam_default...
/usr/lib/python3/dist-packages/supervisor/options.py:474: UserWarning: Supervisord is running as root and it is searching for its configuration file in default locations (including its current working directory); you probably want to specify a "-c" argument specifying an absolute path to a configuration file for improved security.
  self.warnings.warn(
2022-10-18 04:02:58,190 INFO Included extra file "/etc/supervisor/conf.d/jmwalletd.conf" during parsing
2022-10-18 04:02:58,191 INFO Included extra file "/etc/supervisor/conf.d/nginx.conf" during parsing
2022-10-18 04:02:58,193 INFO Included extra file "/etc/supervisor/conf.d/ob-watcher.conf" during parsing
2022-10-18 04:02:58,193 INFO Included extra file "/etc/supervisor/conf.d/supervisord.conf" during parsing
2022-10-18 04:02:58,194 INFO Included extra file "/etc/supervisor/conf.d/tor.conf" during parsing
2022-10-18 04:02:58,267 INFO Set uid to user 0 succeeded
2022-10-18 04:02:58,587 INFO RPC interface 'supervisor' initialized
2022-10-18 04:02:58,595 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2022-10-18 04:02:58,603 INFO supervisord started with pid 9
2022-10-18 04:02:59,615 INFO spawned: 'jmwalletd' with pid 29
2022-10-18 04:02:59,782 INFO spawned: 'nginx' with pid 30
2022-10-18 04:03:00,165 INFO spawned: 'ob-watcher' with pid 31
2022-10-18 04:03:00,442 INFO spawned: 'tor' with pid 33
2022-10-18 04:03:01,731 INFO success: nginx entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2022-10-18 04:03:01,731 INFO success: tor entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2022-10-18 04:03:10,247 INFO success: jmwalletd entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2022-10-18 04:03:10,263 INFO success: ob-watcher entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2022-10-18 04:04:33,355 INFO exited: ob-watcher (exit status 1; not expected)
2022-10-18 04:04:34,082 INFO spawned: 'ob-watcher' with pid 41
2022-10-18 04:04:37,210 INFO exited: jmwalletd (exit status 1; not expected)
2022-10-18 04:04:38,217 INFO spawned: 'jmwalletd' with pid 43
2022-10-18 04:04:41,336 INFO exited: ob-watcher (exit status 1; not expected)
2022-10-18 04:04:42,340 INFO spawned: 'ob-watcher' with pid 46
2022-10-18 04:04:45,325 INFO exited: jmwalletd (exit status 1; not expected)
2022-10-18 04:04:46,330 INFO spawned: 'jmwalletd' with pid 49
2022-10-18 04:04:51,187 INFO exited: ob-watcher (exit status 1; not expected)
2022-10-18 04:04:53,276 INFO spawned: 'ob-watcher' with pid 52
2022-10-18 04:04:57,304 INFO success: jmwalletd entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2022-10-18 04:05:03,337 INFO success: ob-watcher entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2022-10-18 04:05:07,366 INFO exited: jmwalletd (exit status 1; not expected)
2022-10-18 04:05:08,393 INFO spawned: 'jmwalletd' with pid 56
2022-10-18 04:05:12,566 INFO exited: ob-watcher (exit status 1; not expected)
2022-10-18 04:05:13,574 INFO spawned: 'ob-watcher' with pid 58
2022-10-18 04:05:18,603 INFO success: jmwalletd entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2022-10-18 04:05:23,705 INFO success: ob-watcher entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2022-10-18 04:05:28,715 INFO exited: jmwalletd (exit status 1; not expected)
2022-10-18 04:05:29,757 INFO spawned: 'jmwalletd' with pid 61
2022-10-18 04:05:39,065 INFO exited: ob-watcher (exit status 1; not expected)
2022-10-18 04:05:40,069 INFO success: jmwalletd entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2022-10-18 04:05:40,074 INFO spawned: 'ob-watcher' with pid 64
2022-10-18 04:05:50,211 INFO success: ob-watcher entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2022-10-18 04:05:54,218 INFO exited: jmwalletd (exit status 1; not expected)
2022-10-18 04:05:55,226 INFO spawned: 'jmwalletd' with pid 68
2022-10-18 04:06:01,071 INFO exited: ob-watcher (exit status 1; not expected)
2022-10-18 04:06:02,099 INFO spawned: 'ob-watcher' with pid 70
2022-10-18 04:06:06,113 INFO success: jmwalletd entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2022-10-18 04:06:12,198 INFO success: ob-watcher entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2022-10-18 04:06:17,434 INFO exited: jmwalletd (exit status 1; not expected)
2022-10-18 04:06:18,460 INFO spawned: 'jmwalletd' with pid 74
2022-10-18 04:06:26,469 INFO exited: ob-watcher (exit status 1; not expected)
2022-10-18 04:06:27,483 INFO spawned: 'ob-watcher' with pid 76
2022-10-18 04:06:28,494 INFO success: jmwalletd entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2022-10-18 04:06:37,555 INFO success: ob-watcher entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2022-10-18 04:06:43,644 INFO exited: ob-watcher (exit status 1; not expected)
2022-10-18 04:06:44,652 INFO spawned: 'ob-watcher' with pid 80
2022-10-18 04:06:53,034 INFO exited: jmwalletd (exit status 1; not expected)
2022-10-18 04:06:54,046 INFO spawned: 'jmwalletd' with pid 82
2022-10-18 04:06:55,054 INFO success: ob-watcher entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2022-10-18 04:07:00,227 INFO exited: ob-watcher (exit status 1; not expected)
2022-10-18 04:07:01,250 INFO spawned: 'ob-watcher' with pid 85
2022-10-18 04:07:04,029 INFO exited: jmwalletd (exit status 1; not expected)
2022-10-18 04:07:05,035 INFO spawned: 'jmwalletd' with pid 88
2022-10-18 04:07:11,122 INFO exited: ob-watcher (exit status 1; not expected)
2022-10-18 04:07:12,129 INFO spawned: 'ob-watcher' with pid 92
2022-10-18 04:07:14,907 INFO exited: jmwalletd (exit status 1; not expected)
2022-10-18 04:07:16,915 INFO spawned: 'jmwalletd' with pid 94
2022-10-18 04:07:20,517 INFO exited: ob-watcher (exit status 1; not expected)
2022-10-18 04:07:22,524 INFO spawned: 'ob-watcher' with pid 97
2022-10-18 04:07:24,999 INFO exited: jmwalletd (exit status 1; not expected)
2022-10-18 04:07:28,014 INFO spawned: 'jmwalletd' with pid 100
2022-10-18 04:07:30,755 INFO exited: ob-watcher (exit status 1; not expected)
2022-10-18 04:07:33,764 INFO spawned: 'ob-watcher' with pid 103
2022-10-18 04:07:37,700 INFO exited: jmwalletd (exit status 1; not expected)
2022-10-18 04:07:38,703 INFO gave up: jmwalletd entered FATAL state, too many start retries too quickly
2022-10-18 04:07:42,075 INFO exited: ob-watcher (exit status 1; not expected)
2022-10-18 04:07:43,077 INFO gave up: ob-watcher entered FATAL state, too many start retries too quickly`

**Comment by theborakompanioni** [comment]
> I've looked at the jamdocs page and it doesn't seem to point me in the right direction. I am able to ssh into my Umbrel node and grab the logs from the different docker containers, but there is no /var/log/jam/ directory or /home/< user >/.joinmarket/logs directory.
> 
> Attached are the logs when I run `docker <container id for jam_jam_1> logs` [docker_jam_jam_1_logs.txt](https://github.com/joinmarket-webui/jam/files/9806400/docker_jam_jam_1_logs.txt)

Oh, it seems `jmwalletd` exits after 30s of uptime.. that's something new.

To know what's going on, logs from `jmwalletd` service itself are needed.
There should be a `/var/log/jam` directory _inside_ the container: e.g. on Umbrel you can execute something like:
```sh
docker exec -t jam_jam_1 tail -f /var/log/jam/jmwalletd_stdout.log -n 200
```
and
```sh
docker exec -t jam_jam_1 tail -f /var/log/jam/jmwalletd_stderr.log -n 200
```

> And yes, the error persists after reboot.

Thanks for the report. Definitely something to look into more deeply and fix asap once we know what's the underlying issue!

**Comment by librethanks** [comment]
Attached are the outputs from the two docker commands you gave me:

[stdout.log](https://github.com/joinmarket-webui/jam/files/9816912/stdout.log)
[stderr.log](https://github.com/joinmarket-webui/jam/files/9816913/stderr.log)

I noticed this, but not sure if it is relevant to this issue:
`Exception: JoinMarket currently does not support Bitcoin Core descriptor wallets, use legacy wallet (rpc_wallet_file setting in joinmarket.cfg) instead. See docs/USAGE.md for details.
`

**Comment by theborakompanioni** [comment]
Ahhh :grimacing: 
The infamous descriptor wallet bug - I am very sorry this happened. This bug is still causing troubles, looking forward to when it is finally resolved for everyone.
It is fixed now but caused issues for users installing Jam on Bitcoin v23 with Jam <=v0.10.0.
I am afraid there are some manual steps involved to recover from it.

If you have no funds in your wallet, the easiest is probably to follow these steps: https://github.com/joinmarket-webui/jam/issues/502#issuecomment-1252261532
If you do have some funds, either do not uninstall Jam or [import the wallet after reinstalling Jam via the CLI](https://jamdocs.org/FAQ/#can-i-import-an-existing-wallet).

Please let me know if that worked for you.
If you are having troubles or still have some questions, do not hesitate to reach out again.

**Comment by Hukuma1** [comment]
I'm running into this right after updating Umbrel to latest version.

I have no funds in there currently, and tried to uninstall Jam. But it just hangs. Eventually it tries to start it again, and again, and again. If I restart the Umbrel stack it pulls the Jam container again and I'm back to square one. heh

Is there a manual way to delete and purge it instead of trying to use the GUI?

**Comment by theborakompanioni** [comment]
> I'm running into this right after updating Umbrel to latest version.

Sorry that this is happening. The 404 error is fixed in the next version, but the underlying issue can only be resolved manually atm.

> I have no funds in there currently, and tried to uninstall Jam. But it just hangs. Eventually it tries to start it again, and again, and again. If I restart the Umbrel stack it pulls the Jam container again and I'm back to square one. heh

Have you tried deleting the wallet directory as described above?

> Is there a manual way to delete and purge it instead of trying to use the GUI?

I am sure there are ways to do this without the GUI, but please first try the steps as described. If that won't work, we'll find a way!
Even if you have no funds in your wallet, make sure you have all seeds and backups in place :pray:

**Comment by Hukuma1** [comment]
I don't have `bitcoin-cli` installed by default from Umbrel? Is that something I need to install separately? 

```
root@Bitcoin:/opt/umbrel/scripts# bitcoin-cli unloadwallet jam_default
bash: bitcoin-cli: command not found
```

I deleted `jam_default` that was in `/opt/umbrel/app-data/bitcoin/data/bitcoin/jam_default`

Restarted Umbrel and the Jam app is being re-installed automatically (icon still on my dashboard) and still fails. Even without the `jam_default` folder.

**Comment by theborakompanioni** [comment]
Hmm.. does not seem Jam related at first sight.. similar to https://github.com/getumbrel/umbrel/issues/463

I guess you can try something like `sudo ./app uninstall jam` on the CLI to get more information. Have you tried asking in the Umbrel support groups?

**Comment by Hukuma1** [comment]
`./app uninstall jam`

It removes it (`docker ps -a` doesn't show it anymore), but I do get this error, and it doesn't leave the home screen. And if I restart Umbrel, it pulls it automatically again...

```
 ⠿ Image ghcr.io/joinmarket-webui/jam-standalone:v0.1.1-clientserver-v0.9.8@sha256:0b17e193c12b44323e8a815994457f69e0e8ee1299702e6fca7e8547352eafea  Removed 0.9s
 ⠏ Image getumbrel/app-proxy:v0.5.2@sha256:a2e3e0ddfcf84838bf0ba66f4b839ec958832d51f0ac9ace47962459c838b2d6                                          Removing 0.9s
Error response from daemon: conflict: unable to remove repository reference "getumbrel/tor:0.4.7.8@sha256:2ace83f22501f58857fa9b403009f595137fa2e7986c4fda79d82a8119072b6a" (must force) - container de0323045f1d is using its referenced image b3f3dff13ac2
```

Rebooting Umbrel did nothing. This is pretty bizarre and annoying. :/

Guess I'll reach out to Umbrel next. Thanks for your help!

**Comment by theborakompanioni** [comment]
This should be fixed with version v0.1.3 - closing the issue!
Feel free to reopen it if you think this is a mistake.
